### ISSUE TITLE
WebDav Nextcloud remote reporting quota-available-bytes and quota-used-bytes as 0

### ISSUE THREAD

**Issue by mihakrumpestar** [rclone/rclone#7073]
#### The associated forum post URL from `https://forum.rclone.org`
Associated with a merged solution for a bug [https://github.com/rclone/rclone/issues/4348](https://github.com/rclone/rclone/issues/4348) which is similar to this.

#### What is the problem you are having with rclone?
Rclone reports 0 used and 0 available space from WebDav Nextcloud. Even if I use the flag `--vfs-disk-space-total-size 1T` the total space then is 1T (tested on Windows 11), but available space is still 0. GUI programs fail to write to it as they check for available space, while CLI file commands like `cp` and `mv` work fine.

#### What is your rclone version (output from `rclone version`)

```
[REDACTED ~]$ rclone version
rclone v1.62.2
- os/version: arch "rolling" (64 bit)
- os/kernel: 6.3.8-arch1-1 (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.20.4
- go/linking: dynamic
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Arch Linux and Windows 11. Happens on both.

#### Which cloud storage system are you using?

WebDav Nextcloud. I host one myself, the other is a [Hetzner Storage Share](https://www.hetzner.com/storage/storage-share).

#### The command you were trying to run?

Just:

```
rclone mount nextcloud-personal:
```

I use it as a systemd mount (as instructed in Rclone documentation):

```
[Unit]
After=network-online.target
[Mount]
Type=rclone
What=nextcloud-personal:
Where=/mnt/nextcloud_personal
Options=rw,allow_other,args2env,vfs-cache-mode=full,config=/etc/rclone.conf,cache-dir=/var/rclone
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```
[REDACTED ~]$ rclone about nextcloud-personal:  -vv --dump bodies
<7>DEBUG : rclone: Version "v1.62.2" starting with parameters ["rclone" "about" "nextcloud-personal:" "-vv" "--dump" "bodies"]
<7>DEBUG : rclone: systemd logging support activated
<7>DEBUG : Creating backend with remote "nextcloud-personal:"
<7>DEBUG : Using config file from "/home/REDACTED/.config/rclone/rclone.conf"
<7>DEBUG : found headers: 
<7>DEBUG : You have specified to dump information. Please be noted that the Accept-Encoding as shown may not be correct in the request and the response may not show Content-Encoding if the go standard libraries auto gzip encoding was in effect. In this case the body of the request will be gunzipped before showing it.
<7>DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
<7>DEBUG : HTTP REQUEST (req 0xc000a78400)
<7>DEBUG : PROPFIND /remote.php/dav/files/REDACTED/ HTTP/1.1
Host: REDACTED
User-Agent: rclone/v1.62.2
Content-Length: 139
Authorization: XXXX
Depth: 0
Referer: https://REDACTED/remote.php/dav/files/REDACTED/
Accept-Encoding: gzip

<?xml version="1.0" ?>
<D:propfind xmlns:D="DAV:">
 <D:prop>
  <D:quota-available-bytes/>
  <D:quota-used-bytes/>
 </D:prop>
</D:propfind>
<7>DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
<7>DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
<7>DEBUG : HTTP RESPONSE (req 0xc000a78400)
<7>DEBUG : HTTP/2.0 207 Multi-Status
Cache-Control: no-store, no-cache, must-revalidate
Content-Security-Policy: default-src 'none';
Content-Type: application/xml; charset=utf-8
Date: Wed, 21 Jun 2023 17:39:57 GMT
Dav: 1, 3, extended-mkcol, access-control, calendarserver-principal-property-search, nextcloud-checksum-update, nc-calendar-search, nc-enable-birthday-calendar, 2
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Pragma: no-cache
Referrer-Policy: no-referrer
Server: Apache/2.4.54 (Debian)
SameSite=strict
Strict-Transport-Security: 15552000
Vary: Brief,Prefer
X-Content-Type-Options: nosniff
X-Debug-Token: ED3B77mHnoUCxoNcnnAs
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
X-Powered-By: PHP/8.1.12
X-Request-Id: ED3B77mHnoUCxoNcnnAs
X-Robots-Tag: none
X-Xss-Protection: 1; mode=block

<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns"><d:response><d:href>/remote.php/dav/files/REDACTED/</d:href><d:propstat><d:prop><d:quota-available-bytes>0</d:quota-available-bytes><d:quota-used-bytes>0</d:quota-used-bytes></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>
<7>DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Total:   0 B
Used:    0 B
Free:    0 B
<7>DEBUG : 3 go routines active
```

Running `df` also returns the same:

```
[REDACTED ~]$ df -h -T /mnt/nextcloud_personal
Filesystem          Type         Size  Used Avail Use% Mounted on
nextcloud-personal: fuse.rclone     0     0     0    - /mnt/nextcloud_personal
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by kapitainsky** [comment]
Try to add:

`--vfs-used-is-size rclone size           Use the rclone size algorithm for Used size`

to your mount command.

https://rclone.org/commands/rclone_mount/#alternate-report-of-used-bytes

**Comment by mihakrumpestar** [comment]
> Try to add:
> 
> `--vfs-used-is-size rclone size Use the rclone size algorithm for Used size`
> 
> to your mount command.
> 
> https://rclone.org/commands/rclone_mount/#alternate-report-of-used-bytes

I tried it with:

```
[Unit]
After=network-online.target
[Mount]
Type=rclone
What=nextcloud-personal:
Where=/mnt/nextcloud_personal
Options=rw,allow_other,args2env,vfs-cache-mode=full,vfs-used-is-size=true,vfs-disk-space-total-size=1T,config=/etc/rclone.conf,cache-dir=/var/rclone
```

But it still reports 0 bytes free:

```
[ ~]$ df -h -T /mnt/nextcloud_personal
Filesystem                 Type             Size  Used Avail Use% Mounted on
nextcloud-personal: fuse.rclone  1,0T  1,0T   0       100% /mnt/nextcloud_personal
```

**Comment by kapitainsky** [comment]
> I tried it with:

No you did not. This is the flag to be used:

`--vfs-used-is-size rclone size Use the rclone size algorithm for Used size`

**Comment by mihakrumpestar** [comment]
> 

Is `vfs-used-is-size=true` not the right one?

**Comment by kapitainsky** [comment]
https://rclone.org/commands/rclone_mount/#options

**Comment by mihakrumpestar** [comment]
> https://rclone.org/commands/rclone_mount/#options

I don't understand what exactly should be the value of the flag (if that is the problem), but as the error indicates, it should be a boolean:
![image](https://github.com/rclone/rclone/assets/70652456/092cf396-0643-4352-be42-e3c11d7dbb28)

Am I missing something? The mount does take some time to be available (about 20 seconds instead of instantly like before), so I think the rclone does try to calculate the actual used size.

**Comment by kapitainsky** [comment]
Yes I am looking into it now - something is not right but I am not sure what yet.

**Comment by mihakrumpestar** [comment]
Based on the logs I think that Nextcloud sends all "usage/disk size" parameters as 0, and Rclone displays them as that. The only fix I can think of is a flag or logic, that would make Rclone aware, that all zeroes can't be right and that remote is sending the wrong "size" data.

**Comment by ncw** [comment]
I can't replicate the 0 bytes entries  on nextcloud or on a hetzner storage box.

```
$ rclone-v1.62.2 about hbox-webdav:
$ rclone-v1.62.2 about nextcloud:
Used:    35.916 MiB
```

Your server is definitely returning 0s though

```xml
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns">
<d:response>
<d:href>/remote.php/dav/files/REDACTED/</d:href>
<d:propstat>
<d:prop>
<d:quota-available-bytes>0</d:quota-available-bytes>
<d:quota-used-bytes>0</d:quota-used-bytes>
</d:prop>
<d:status>HTTP/1.1 200 OK</d:status>
</d:propstat>
</d:response>
</d:multistatus>
```

So I'd suggest this is misconfiguration in the sever rather than a bug in rclone.

What does the hetzner storage box return to `rclone about storagebox: -vv --dump bodies`?

Can you show your config for both please, redacting any secrets.

Thanks

**Comment by mihakrumpestar** [comment]
I forgot to mention that we have opted for the latest version of Nextcloud on Hetzner storage box due to the standard one being too old for us. The current version is therefore `Nextcloud Hub 4, 26.0.3`. At home, I have version `25.0.1` and I am observing the same behavior. 

All commands below are run on Windows Server 2022 Standard against the Hetzner storage box running `Nextcloud 26.0.3`.

This is the dump from Windows:
```powershell
PS C:\Users\REDACTED> rclone about nextcloud: -vv --dump bodies
2023/07/12 23:00:36 DEBUG : rclone: Version "v1.62.2" starting with parameters ["C:\\ProgramData\\chocolatey\\lib\\rclone.portable\\tools\\rclone-v1.62.2-windows-amd64\\rclone.exe" "about" "nextcloud:" "-vv" "--dump" "bodies"]
2023/07/12 23:00:36 DEBUG : Creating backend with remote "nextcloud:"
2023/07/12 23:00:36 DEBUG : Using config file from "C:\\Users\\REDACTED\\AppData\\Roaming\\rclone\\rclone.conf"
2023/07/12 23:00:36 DEBUG : found headers:
2023/07/12 23:00:36 DEBUG : You have specified to dump information. Please be noted that the Accept-Encoding as shown may not be correct in the request and the response may not show Content-Encoding if the go standard libraries auto gzip encoding was in effect. In this case the body of the request will be gunzipped before showing it.
2023/07/12 23:00:36 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2023/07/12 23:00:36 DEBUG : HTTP REQUEST (req 0xc000822800)
2023/07/12 23:00:36 DEBUG : PROPFIND /remote.php/dav/files/REDACTED/ HTTP/1.1
Host: REDACTED
User-Agent: rclone/v1.62.2
Content-Length: 139
Authorization: XXXX
Depth: 0
Referer: https://REDACTED/remote.php/dav/files/REDACTED/
Accept-Encoding: gzip

<?xml version="1.0" ?>
<D:propfind xmlns:D="DAV:">
 <D:prop>
  <D:quota-available-bytes/>
  <D:quota-used-bytes/>
 </D:prop>
</D:propfind>
2023/07/12 23:00:36 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2023/07/12 23:00:36 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2023/07/12 23:00:36 DEBUG : HTTP RESPONSE (req 0xc000822800)
2023/07/12 23:00:36 DEBUG : HTTP/2.0 207 Multi-Status
Cache-Control: no-store, no-cache, must-revalidate
Content-Security-Policy: default-src 'none';
Content-Type: application/xml; charset=utf-8
Date: Wed, 12 Jul 2023 21:00:36 GMT
Dav: 1, 3, extended-mkcol, access-control, calendarserver-principal-property-search, nextcloud-checksum-update, nc-calendar-search, nc-enable-birthday-calendar
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Pragma: no-cache
Referrer-Policy: no-referrer
Server: openresty
Set-Cookie: oc_sessionPassphrase=REDACTED; path=/; secure; HttpOnly; SameSite=Lax
Set-Cookie: __Host-nc_sameSiteCookielax=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=lax
Set-Cookie: __Host-nc_sameSiteCookiestrict=true; path=/; httponly;secure; expires=Fri, 31-Dec-2100 23:59:59 GMT; SameSite=strict
Set-Cookie: REDACTED; path=/; secure; HttpOnly; SameSite=Lax
Vary: Brief,Prefer
X-Content-Type-Options: nosniff
X-Debug-Token: REDACTED
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
X-Request-Id: REDACTED
X-Robots-Tag: noindex, nofollow
X-Xss-Protection: 1; mode=block

<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns"><d:response><d:href>/remote.php/dav/files/REDACTED/</d:href><d:propstat><d:prop><d:quota-available-bytes>0</d:quota-available-bytes><d:quota-used-bytes>0</d:quota-used-bytes></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat></d:response></d:multistatus>
2023/07/12 23:00:36 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Total:   0 B
Used:    0 B
Free:    0 B
2023/07/12 23:00:36 DEBUG : 3 go routines active
```

Command I use:
```powershell
mount nextcloud:/ \\server\nextcloud\ --vfs-cache-mode writes --file-perms 0640 --dir-perms 0640 --log-file C:\Users\REDACTED\Desktop\rclone.log
```

The command is run as a service using NSSM (the same result regardless if run directly from terminal).

Here is the config:
```powershell
[nextcloud]
type = webdav
url = https://REDACTED/remote.php/dav/files/REDACTED/
vendor = nextcloud
user = REDACTED
pass = REDACTED
```

**Comment by mihakrumpestar** [comment]
I just had the `eureka` moment. I remembered I have probably more unique setup on Nextcloud than most people, as every user has either mapped a remote storage or shared/group storage. None of them have any permission to use the actual default Nextcloud storage, as I have set default storage quota to `0 B` to all of them (I just tested that seting this to anything else works properly with Rclone now). This way they can freely use the storage I provide, without being able to spam anything else.

Nextcloud does allow this, and it works, but Rclone does not have a way to comprehend this use case. Basically, users home directory has 0 space, so he is not able to save anything there, but he has a Documents and Downloads folder as external storage where he has 1 TB of space.

The question now is if there is any way to support this or allow overriding the disk used size in Rclone (there already is an option to override the disk space total)?
